neqo-http3/benches/streams.rs

```diff
@@ -76,21 +76,18 @@ fn connect() -> (Http3Client, Http3Server) {
 fn criterion_benchmark(c: &mut Criterion) {
     fixture_init();
 
-    for (streams, data_size) in [
-        (1, 1),
-        (1000, 1),
-        (10000, 1),
-        (1, 1000),
-        (100, 1000),
-        (1000, 1000),
-    ] {
+    for (streams, data_size) in [(1_000, 1), (1_000, 1_000)] {
         let mut group = c.benchmark_group(format!("{streams} streams of {data_size} bytes"));
+
+        // High variance benchmark. Increase default sample size (100).
+        group.sample_size(500);
+
         group.bench_function("multistream", |b| {
             let data = vec![0; data_size];
             b.iter_batched_ref(
                 connect,
                 |(client, server)| use_streams(client, server, streams, &data),
-                BatchSize::PerIteration,
+                BatchSize::SmallInput,
```

Comment:
I read the documentation of `BatchSize` as such, that one should only deviate from the default (i.e. `SmallInput`) on memory pressure issues. While `client` and `server` are likely large, I doubt they put the machine under memory pressure. Thus I suggest sticking with the default.

I don't feel strongly about it. Happy to revert.

https://docs.rs/criterion/latest/criterion/enum.BatchSize.html#variant.SmallInput